Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+        if dataclass_field:
+            d[attr_name] = attr_value
+
+    d = flatten_xml_to_dict(element, d, class_type)
+
+    return class_type(**d)
+
+
+def resolve_host_id(device: PanDevice):
+    """
+    Gets the ID of the host from a PanDevice object. This may be an IP address or serial number.
+    """
+    host_id: str = ""
+    if device.hostname:
+        host_id = device.hostname
+    if device.serial:
+        host_id = device.serial
+
+    return host_id
+
+
+def resolve_container_name(container: Union[Panorama, Firewall, DeviceGroup, Template, Vsys]):
+    """
+    Gets the name of a given PanDevice container or if it's not a container, returns shared.
+    """
+    if type(container) in [Panorama, Firewall]:
+        return "shared"
+
+    return container.name
+
+
+class PanoramaCommand:
+    """Commands that can only be run, or are relevant only on Panorama."""
+    GET_DEVICEGROUPS_COMMAND = "show devicegroups"
+    GET_TEMPLATE_STACK_COMMAND = "show template-stack"
+
+    @staticmethod
+    def get_device_groups(topology: Topology, device_filter_str: str = None) -> List[DeviceGroupInformation]:
+        """Get all the device groups from Panorama and their associated devices."""
+        device: Panorama
+        result = []
+        for device in topology.active_top_level_devices(device_filter_str):
+            if type(device) is Panorama:
+                response = run_op_command(device, PanoramaCommand.GET_DEVICEGROUPS_COMMAND)
+                for device_group_xml in response.findall("./result/devicegroups/entry"):
+                    dg_name = get_element_attribute(device_group_xml, "name")
+                    for device_xml in device_group_xml.findall("./devices/entry"):
+                        r: DeviceGroupInformation = dataclass_from_element(device, DeviceGroupInformation,
+                                                                           device_xml)
+                        r.name = dg_name
+                        result.append(r)
+
+        return result
+
+    @staticmethod
+    def get_template_stacks(topology: Topology, device_filter_str: str = None) -> List[
+        TemplateStackInformation]:
+        """Get all the template-stacks from Panorama and their associated devices."""
+
+        device: Panorama
+        result = []
+        for device in topology.active_top_level_devices(device_filter_str):
+            if isinstance(device, Panorama):
+                response = run_op_command(device, PanoramaCommand.GET_TEMPLATE_STACK_COMMAND)
+                for template_stack_xml in response.findall("./result/template-stack/entry"):
+                    template_name = get_element_attribute(template_stack_xml, "name")
+                    for device_xml in template_stack_xml.findall("./devices/entry"):
+                        r: TemplateStackInformation = dataclass_from_element(device, TemplateStackInformation,
```

Comment:
please give meaningful variable name

Reply:
This, and other occurences of this, should be fixed now.